Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -362,32 +378,86 @@ extension _FileManagerImpl {
         isDirectory = result.isDirectory
         return true
     }
-    
+
+#if !os(Windows)
     private func _fileAccessibleForMode(_ path: String, _ mode: Int32) -> Bool {
         path.withFileSystemRepresentation { ptr in
             guard let ptr else { return false }
             return access(ptr, mode) == 0
         }
     }
-    
+#endif
+
     func isReadableFile(atPath path: String) -> Bool {
+#if os(Windows)
+        do {
+            return try path.withNTPathRepresentation {
+                var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+                return GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) == true
+            }
+        } catch {
+            return false
+        }
+#else
         _fileAccessibleForMode(path, R_OK)
+#endif
     }
     
     func isWritableFile(atPath path: String) -> Bool {
+#if os(Windows)
+        do {
+            return try path.withNTPathRepresentation {
+                var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+                guard GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) else {
+                    return false
+                }
+                return faAttributes.dwFileAttributes & DWORD(FILE_ATTRIBUTE_READONLY) != DWORD(FILE_ATTRIBUTE_READONLY)
+            }
+        } catch {
+            return false
+        }
+#else
         _fileAccessibleForMode(path, W_OK)
+#endif
     }
     
     func isExecutableFile(atPath path: String) -> Bool {
+#if os(Windows)
+        do {
+            return try path.withNTPathRepresentation {
+                var dwBinaryType: DWORD = 0
+                return GetBinaryTypeW($0, &dwBinaryType)
+            }
+        } catch {
+            return false
+        }
+#else
         _fileAccessibleForMode(path, X_OK)
+#endif
     }
     
     func isDeletableFile(atPath path: String) -> Bool {
         var parent = path.deletingLastPathComponent()
         if parent.isEmpty {
             parent = fileManager.currentDirectoryPath
         }
-        
+
+#if os(Windows)
+        var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+        guard (try? parent.withNTPathRepresentation({
+            GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes)
+        })) == true else { return false }
+        if faAttributes.dwFileAttributes & DWORD(FILE_ATTRIBUTE_READONLY) == DWORD(FILE_ATTRIBUTE_READONLY) {
+            return false
+        }
+        guard (try? path.withNTPathRepresentation({
+            GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes)
+        })) == true else { return false }
+        if faAttributes.dwFileAttributes & DWORD(FILE_ATTRIBUTE_READONLY) == DWORD(FILE_ATTRIBUTE_READONLY) {
+            return false
+        }
+        return true
```

Comment:
To reuse code, could all of this just be

```swift
fileManager.isWritableFile(atPath: parent) && fileManager.isWritableFile(atPath: path)
```

That'd be a lot simpler / more readable if I'm write that it's the same behavior